pkg/disk/disk_windows.go

```diff
@@ -5,8 +5,169 @@
 
 package disk
 
+import (
+	"bytes"
+	_ "embed"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/afero"
+
+	"github.com/runfinch/finch/pkg/winutil"
+)
+
 // EnsureUserDataDisk checks the current disk configuration and fixes it if needed.
 // on Windows, this is a no-op.
 func (m *userDataDiskManager) EnsureUserDataDisk() error {
+	diskPath := m.finch.UserDataDiskPath(m.rootDir)
+	disksDir := filepath.Dir(diskPath)
+
+	m.logger.Debugf("diskPath: %s", diskPath)
+	m.logger.Debugf("disksDir: %s", disksDir)
+
+	if _, err := m.fs.Stat(disksDir); errors.Is(err, fs.ErrNotExist) {
+		if err := m.fs.MkdirAll(disksDir, 0o755); err != nil {
+			return fmt.Errorf("could not create persistent disk directory: %w", err)
+		}
+	}
+
+	if _, err := m.fs.Stat(diskPath); errors.Is(err, fs.ErrNotExist) {
+		if err := m.createDisk(diskPath); err != nil {
+			return fmt.Errorf("could not create persistent disk: %w", err)
+		}
+	}
+
+	if err := m.attachDisk(diskPath); err != nil {
+		return fmt.Errorf("could not attach persistent disk: %w", err)
+	}
+
+	return nil
+}
+
+// DetachUserDataDisk unmounts the disk in wsl.
+func (m *userDataDiskManager) DetachUserDataDisk() error {
+	cmd := m.ecc.Create(
+		"wsl.exe",
+		"--unmount",
+		`\\?\`+m.finch.UserDataDiskPath(m.rootDir),
+	)
+
+	m.logger.Debugf("running attach cmd: %v", cmd)
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to attach disk: %w, command output: %s", err, out)
+	}
+
+	return nil
+}
+
+func (m *userDataDiskManager) diskExists(diskPath string) (bool, error) {
+	disksDir := filepath.Dir(diskPath)
+	_, err := m.fs.Stat(filepath.Dir(diskPath))
+	if errors.Is(err, fs.ErrNotExist) {
+		if err := m.fs.MkdirAll(disksDir, 0o755); err != nil {
+			return false, fmt.Errorf("could not create persistent disk directory: %w", err)
+		}
+	}
+
+	return true, nil
+}
+
+//go:embed createDiskAdmin.TEMPLATE.ps1
+var createDiskTmpl string
```

Comment:
Is this used?

Reply:
Not anymore, since the logic is moved to another package. I'll remove it 👍

Reply:
Removed in latest revision